nerdlandbot/commands/pets.py

```diff
@@ -0,0 +1,309 @@
+        name="add_pet",
+        brief="add_pet_brief",
+        help="add_pet_help",
+        usage="add_pet_usage",
+    )
+    @commands.guild_only()
+    async def add_pet(
+        self,
+        ctx: commands.Context,
+        name: typing.Optional[str] = None,
+        category: typing.Optional[str] = None,
+    ):
+        lang, guild_data = await asyncio.gather(
+            culture(ctx),
+            get_guild_data(ctx.message.guild.id),
+        )
+        if name is None:
+            message = translate("pet_need_name", lang)
+            return await ctx.send(message)
+        if category is None:
+            message = translate("pet_need_category", lang)
+            return await ctx.send(message)
+
+        if category.lower() not in guild_data.pets_categories:
+            message = translate("pet_category_nonexistant", lang)
+            return await ctx.send(message)
+
+        
+        await guild_data.add_pet(name, ctx.author.id, category.lower())
+        pet_id = guild_data.pets_last_id
+        if ctx.message.attachments:
+            await resize_and_save(ctx, ctx.message.attachments[0], pet_id)
+            msg = translate("pet_added_succes", lang).format(
+                name.capitalize(), pet_id)
+            await ctx.send(msg)
+        else:
+            embed_title = translate("pet_embed_title", lang)
+            question = translate("waiting_question",
+                                 lang).format(name.capitalize())
+            embed = discord.Embed(
+                title=embed_title,
+                description=question,
+            )
+            await ctx.send(embed=embed)
+            try:
+                reaction = await ctx.bot.wait_for(
+                    "message", timeout=INTERACT_TIMEOUT, check=check(ctx.author)
+                )
+
+                if reaction.content == "0":
+                    await guild_data.delete_pet(pet_id)
+                    abort = translate("command_abort", lang)
+                    embed = discord.Embed(
+                        title=embed_title,
+                        description=abort,
+                    )
+                    return await ctx.send(embed=embed)
+
+                if reaction.attachments:
+                    await resize_and_save(ctx, reaction.attachments[0], pet_id)
+                    msg = translate("pet_added_succes", lang).format(
+                        name.capitalize(), pet_id
+                    )
+                    await ctx.send(msg)
+                else:
+                    msg = translate("pet_reaction_error", lang)
+                    await ctx.send(msg)
+
+            except asyncio.TimeoutError:
+                await guild_data.delete_pet(pet_id)
+                timeout = translate("command_timeout", lang)
+                embed = discord.Embed(
+                    title=embed_title,
+                    description=timeout,
+                )
+                return await ctx.send(embed=embed)
```

Comment:
Nitpick, but this could perhaps be in a finally statement. Then in case of any errors, the pet would also be deleted.

Reply:
The pet doesn't need to be deleted if there is no error though. It only has to be when there is a timeout error, when the user didn't respond with a picture.